src/Chapter3.hs

```diff
@@ -970,6 +1083,31 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
+data Weekday
+    = Monday
+    | Tuesday
+    | Wednesday
+    | Thursday
+    | Friday
+    | Saturday
+    | Sunday deriving (Show, Eq, Ord, Bounded, Enum)
+
+isWeekend :: Weekday -> Bool
+isWeekend weekday = case weekday of
+    Saturday -> True
+    Sunday   -> True
+    _        -> False
+
+nextDay :: Weekday -> Weekday
+nextDay weekday =
+    if weekday == maxBound
+    then minBound
+    else succ weekday
+
+daysToParty :: Weekday -> Int
+daysToParty weekday
+    | weekday <= Friday = fromEnum Friday - fromEnum weekday
+    | otherwise = fromEnum Sunday - (fromEnum weekday - fromEnum Friday) + 1
```

Comment:
Great catch on the corner case! 👁️ 

You could use the fact that we know the exact number of days, that means, that we could just take the subtraction by `mod .. 7` 🥳